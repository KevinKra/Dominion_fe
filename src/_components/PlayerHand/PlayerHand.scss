@import "../../styles/_variables.scss";

.PlayerHand {
  display: flex;
  justify-content: center;
  border: 1px solid blue;
  .card:hover {
    transition: 0.8s transform ease;
    transform: translateY(-60%);
  }
}

@media screen and (max-width: 1500px) {
  .PlayerHand {
    // width: 50%;
    transform: translateY(60%);
  }
}
