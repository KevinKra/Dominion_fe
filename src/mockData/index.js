export const mockActionCard = {
  name: "Village",
  tags: ["+1 Card", "+2 Actions"],
  desc: "",
  category: ["Action"],
  cost: 3,
  spendingPower: 0,
  actionsProvided: 2,
  cardsToDraw: 1,
  image: "Village.jpg",
  id: 1231
};

export const mockTreasureCard = {
  name: "Copper",
  category: ["Treasure"],
  cost: 0,
  image: "Copper.jpg",
  spendingPower: 1,
  id: 1113
};

export const mockVictoryCard = {
  name: "Province",
  category: ["Victory"],
  cost: 8,
  image: "Province.jpg",
  victoryPoints: 6,
  id: 3781
};

export const mockGameState = {
  tableDeck: [
    {
      name: "Gold",
      category: ["Money"],
      cost: 6,
      victoryPoints: 0,
      spendingPower: 3,
      buyingPower: 0,
      actionsProvided: 3,
      cardsToDraw: 0,
      image: "./gold.jpg",
      desc: "",
      tags: [],
      countAvailable: 5,
      id_list: [1011, 1012, 1013, 1014, 1015]
    },
    {
      name: "Estate",
      category: ["Victory"],
      cost: 2,
      victoryPoints: 1,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "./estate.jpg",
      desc: "",
      tags: [],
      countAvailable: 8,
      id_list: [1041, 1042, 1043, 1044, 1045, 1046, 1047, 1048]
    }
  ],
  playerOrder: ["Player_1_Name", "Player_2_Name"],
  playerInfo: {
    Player_1_Name: {
      deckSize: 10,
      topCardDiscard: 0,
      handSize: 5
    },
    Player_2_Name: {
      deckSize: 10,
      topCardDiscard: 0,
      handSize: 5
    }
  },
  activePlayerName: "Player_1_Name",
  activePlayerId: 1,
  cards: [
    {
      name: "Village",
      tags: ["+1 Card", "+2 Actions"],
      desc: "",
      category: ["Action"],
      cost: 3,
      spendingPower: 0,
      actionsProvided: 2,
      cardsToDraw: 1,
      image: "Village.jpg",
      id: 1231
    },
    {
      name: "Militia",
      tags: ["+2 Gold"],
      desc: "Each other player discards down to 3 cards in his hand.",
      category: ["Action", "Attack"],
      cost: 4,
      spendingPower: 2,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "Militia.jpg",
      id: 1391
    },
    {
      name: "Smithy",
      tags: ["+3 Cards"],
      desc: "",
      category: ["Action"],
      cost: 4,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 3,
      image: "Smithy.jpg",
      id: 9531
    },
    {
      name: "Market",
      tags: ["+1 Card", "+1 Action", "+1 Buy", "+1 Gold"],
      desc: "",
      category: ["Action"],
      cost: 5,
      spendingPower: 1,
      buyingPower: 1,
      actionsProvided: 1,
      cardsToDraw: 1,
      image: "Market.jpg",
      id: 1532
    },
    {
      name: "Mine",
      tags: [],
      desc:
        "Trash a treasure Card from your hand. Gain a treasure card costing up to 3 treasure more; put it in your hand.",
      category: ["Action"],
      cost: 5,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "Mine.jpg",
      id: 1522
    },
    {
      name: "Remodel",
      tags: [],
      desc:
        "Trash a card from your hand. Gain a card costing up to 2 treasure more than the trashed card.",
      category: ["Action"],
      cost: 4,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "Remodel.jpg",
      id: 3333
    },
    {
      name: "Cellar",
      tags: ["+1 Action"],
      desc: "Discard any number of cards. +1 Card per card discarded.",
      category: ["Action"],
      cost: 2,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 1,
      cardsToDraw: 0,
      image: "Cellar.jpg",
      id: 9831
    },
    {
      name: "Moat",
      tags: ["+2 Cards"],
      desc:
        "When another player plays an Attack card, you may reveal this from your hand. If you do, you are unaffected by that attack.",
      category: ["Action", "Reaction"],
      cost: 2,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 2,
      image: "Moat.jpg",
      id: 1938
    },
    {
      name: "WoodCutter",
      tags: ["+1 Buy", "+2 Gold"],
      desc: "",
      category: ["Action"],
      cost: 3,
      spendingPower: 2,
      buyingPower: 1,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "WoodCutter.jpg",
      id: 9781
    },
    {
      name: "WorkShop",
      tags: [],
      desc: "Gain a card costing up to 4 treasure.",
      category: ["Action"],
      cost: 3,
      spendingPower: 0,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "Workshop.jpg",
      id: 1333
    },
    {
      name: "Gold",
      category: ["Treasure"],
      cost: 6,
      image: "Gold.jpg",
      spendingPower: 3,
      id: 1515
    },
    {
      name: "Silver",
      category: ["Treasure"],
      cost: 3,
      image: "Silver.jpg",
      spendingPower: 2,
      id: 8923
    },
    {
      name: "Copper",
      category: ["Treasure"],
      cost: 0,
      image: "Copper.jpg",
      spendingPower: 1,
      id: 1113
    },
    {
      name: "Estate",
      category: ["Victory"],
      cost: 2,
      image: "Estate.jpg",
      victoryPoints: 1,
      id: 6123
    },
    {
      name: "Duchy",
      category: ["Victory"],
      cost: 5,
      image: "Duchy.jpg",
      victoryPoints: 3,
      id: 1344
    },
    {
      name: "Province",
      category: ["Victory"],
      cost: 8,
      image: "Province.jpg",
      victoryPoints: 6,
      id: 3781
    }
  ]
};

export const mockPlayerData = {
  playerId: 1,
  deck: [
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    },
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    },
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    },
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    },
    {
      name: "Copper",
      category: ["Money"],
      cost: 0,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 0,
      actionsProvided: 0,
      cardsToDraw: 0,
      image: "copper.jpg",
      desc: "",
      tags: []
    }
  ],
  discard: [
    {
      name: "Market",
      category: "Action",
      cost: 5,
      victoryPoints: 0,
      spendingPower: 1,
      buyingPower: 1,
      actionsProvided: 1,
      cardsToDraw: 1,
      image: "market.jpg",
      desc: "",
      tags: ["+1 Card", "+1 Action", "+1 Buy", "+1 Gold"]
    }
  ]
};

// const boughtCards = [
//   { name: "Village", amountBought: 2 },
//   { name: "Mine", amountBought: 1 }
// ];
