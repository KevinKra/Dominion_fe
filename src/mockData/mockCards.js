export const cards = [
  {
    name: "Village",
    tags: ["+1 Card", "+2 Actions"],
    desc: "",
    type: ["Action"],
    cost: 3,
    spendingPower: 0,
    actionsProvided: 2,
    cardsToDraw: 1,
    image: "Village.jpg",
    id: 1231
  },
  {
    name: "Militia",
    tags: ["+2 Gold"],
    desc: "Each other player discards down to 3 cards in his hand.",
    type: ["Action", "Attack"],
    cost: 4,
    spendingPower: 2,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "Militia.jpg",
    id: 1391
  },
  {
    name: "Smithy",
    tags: ["+3 Cards"],
    desc: "",
    type: ["Action"],
    cost: 4,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 3,
    image: "Smithy.jpg",
    id: 9531
  },
  {
    name: "Market",
    tags: ["+1 Card", "+1 Action", "+1 Buy", "+1 Gold"],
    desc: "",
    type: ["Action"],
    cost: 5,
    spendingPower: 1,
    buyingPower: 1,
    actionsProvided: 1,
    cardsToDraw: 1,
    image: "Market.jpg",
    id: 1532
  },
  {
    name: "Mine",
    tags: [],
    desc:
      "Trash a treasure Card from your hand. Gain a treasure card costing up to 3 treasure more; put it in your hand.",
    type: ["Action"],
    cost: 5,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "Mine.jpg",
    id: 1522
  },
  {
    name: "Remodel",
    tags: [],
    desc:
      "Trash a card from your hand. Gain a card costing up to 2 treasure more than the trashed card.",
    type: ["Action"],
    cost: 4,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "Remodel.jpg",
    id: 3333
  },
  {
    name: "Cellar",
    tags: ["+1 Action"],
    desc: "Discard any number of cards. +1 Card per card discarded.",
    type: ["Action"],
    cost: 2,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 1,
    cardsToDraw: 0,
    image: "Cellar.jpg",
    id: 9831
  },
  {
    name: "Moat",
    tags: ["+2 Cards"],
    desc:
      "When another player plays an Attack card, you may reveal this from your hand. If you do, you are unaffected by that attack.",
    type: ["Action", "Reaction"],
    cost: 2,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 2,
    image: "Moat.jpg",
    id: 1938
  },
  {
    name: "WoodCutter",
    tags: ["+1 Buy", "+2 Gold"],
    desc: "",
    type: ["Action"],
    cost: 3,
    spendingPower: 2,
    buyingPower: 1,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "WoodCutter.jpg",
    id: 9781
  },
  {
    name: "WorkShop",
    tags: [],
    desc: "Gain a card costing up to 4 treasure.",
    type: ["Action"],
    cost: 3,
    spendingPower: 0,
    buyingPower: 0,
    actionsProvided: 0,
    cardsToDraw: 0,
    image: "Workshop.jpg",
    id: 1333
  },
  {
    name: "Gold",
    type: ["Treasure"],
    cost: 6,
    image: "Gold.jpg",
    spendingPower: 3,
    id: 1515
  },
  {
    name: "Silver",
    type: ["Treasure"],
    cost: 3,
    image: "Silver.jpg",
    spendingPower: 2,
    id: 8923
  },
  {
    name: "Copper",
    type: ["Treasure"],
    cost: 0,
    image: "Copper.jpg",
    spendingPower: 1,
    id: 1113
  },
  {
    name: "Estate",
    type: ["Victory"],
    cost: 2,
    image: "Estate.jpg",
    victoryPoints: 1,
    id: 6123
  },
  {
    name: "Duchy",
    type: ["Victory"],
    cost: 5,
    image: "Duchy.jpg",
    victoryPoints: 3,
    id: 1344
  },
  {
    name: "Province",
    type: ["Victory"],
    cost: 8,
    image: "Province.jpg",
    victoryPoints: 6,
    id: 3781
  }
];

// const boughtCards = [
//   { name: "Village", amountBought: 2 },
//   { name: "Mine", amountBought: 1 }
// ];
