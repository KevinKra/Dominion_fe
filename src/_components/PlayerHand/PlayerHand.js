import React, { Component } from "react";
import { connect } from "react-redux";
import "./PlayerHand.scss";
import TreasureCard from "../TreasureCard/TreasureCard";
import ActionCard from "../ActionCard/ActionCard";
import Card from "../Card/Card";
import { TimelineMax } from "gsap";
import { gsapCardsCycleIn } from "../../greenSock/animations";
import * as actions from "../../_redux/actions/";
import { mockPlayerData } from "../../mockData";

export class PlayerHand extends Component {
  constructor(props) {
    super(props);
    this.cards = [];
    this.tl = new TimelineMax({ paused: true });
    this.state = { renderHand: false };
  }
  componentDidMount() {}

  componentDidUpdate(prevProps, prevState) {
    if (this.props.round === 1 && this.props.playerHand.length === 5) {
      gsapCardsCycleIn(this.tl, this.cards, this.showUserHand);
      this.tl.play();
    }
    // if (prevProps.round !== this.props.round && this.props.playerHand.length) {
    //   gsapCardsCycleIn(this.tl, this.cards, this.showUserHand);
    //   this.tl.play();
    // }
    if (prevProps.playerDeck !== this.props.playerDeck && this.props.playerHand.length) {
      gsapCardsCycleIn(this.tl, this.cards, this.showUserHand);
      this.tl.play();
    }
  }

  showUserHand = () => {
    this.props.triggerHandAnimation();
  };

  render() {
    this.tl
      .kill()
      .clear()
      .pause(0);
    return (
      <React.Fragment>
        <section className='hand-animations'>
          {this.props.playerHand.length &&
            this.props.playerHand.map((card, i) => {
              console.log(card);
              return (
                <div className='animated-card' key={i} ref={div => (this.cards[i] = div)}>
                  <p>{card.name}</p>
                </div>
              );
            })}
        </section>

        <section
          className='PlayerHand'
          style={
            this.props.animateHand
              ? { transform: "translateY(50%)" }
              : { transform: "translateY(100%)" }
          }
        >
          {this.props.animateHand &&
            this.props.playerHand.map(card => {
              if (card.category[0] === "Money") {
                return <TreasureCard card={card} key={card.id} />;
              }
              if (card.category[0] === "Action") {
                return <ActionCard card={card} key={card.id} />;
              }
              if (card.category[0] === "Victory") {
                return (
                  <Card
                    name={card.name}
                    desc={card.desc}
                    category={card.category}
                    image={card.image}
                    cost={card.cost}
                    id={card.id}
                    key={card.id}
                  />
                );
              }
            })}
        </section>
      </React.Fragment>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  triggerHandAnimation: () => dispatch(actions.triggerHandAnimation())
});

export const mapStateToProps = state => ({
  playerHand: state.playerHand,
  playerDeck: state.playerDeck,
  round: state.round,
  animateHand: state.animateHand
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PlayerHand);
