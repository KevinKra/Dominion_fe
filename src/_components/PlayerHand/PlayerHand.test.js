import React from "react";
import { PlayerHand } from "./PlayerHand";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { cards } from "../../mockData";

describe("<PlayerHand />", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<PlayerHand playerHand={cards} />);
  });
  it("should match the snapshot", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
