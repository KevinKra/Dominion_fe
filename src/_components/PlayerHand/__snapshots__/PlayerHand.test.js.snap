// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerHand /> should match the snapshot 1`] = `
<section
  className="PlayerHand"
>
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 2,
        "cardsToDraw": 1,
        "category": Array [
          "Action",
        ],
        "cost": 3,
        "desc": "",
        "id": 1231,
        "image": "Village.jpg",
        "name": "Village",
        "spendingPower": 0,
        "tags": Array [
          "+1 Card",
          "+2 Actions",
        ],
      }
    }
    key="1231"
  />
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 0,
        "buyingPower": 0,
        "cardsToDraw": 0,
        "category": Array [
          "Action",
          "Attack",
        ],
        "cost": 4,
        "desc": "Each other player discards down to 3 cards in his hand.",
        "id": 1391,
        "image": "Militia.jpg",
        "name": "Militia",
        "spendingPower": 2,
        "tags": Array [
          "+2 Gold",
        ],
      }
    }
    key="1391"
  />
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 0,
        "buyingPower": 0,
        "cardsToDraw": 3,
        "category": Array [
          "Action",
        ],
        "cost": 4,
        "desc": "",
        "id": 9531,
        "image": "Smithy.jpg",
        "name": "Smithy",
        "spendingPower": 0,
        "tags": Array [
          "+3 Cards",
        ],
      }
    }
    key="9531"
  />
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 1,
        "buyingPower": 1,
        "cardsToDraw": 1,
        "category": Array [
          "Action",
        ],
        "cost": 5,
        "desc": "",
        "id": 1532,
        "image": "Market.jpg",
        "name": "Market",
        "spendingPower": 1,
        "tags": Array [
          "+1 Card",
          "+1 Action",
          "+1 Buy",
          "+1 Gold",
        ],
      }
    }
    key="1532"
  />
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 0,
        "buyingPower": 0,
        "cardsToDraw": 0,
        "category": Array [
          "Action",
        ],
        "cost": 5,
        "desc": "Trash a treasure Card from your hand. Gain a treasure card costing up to 3 treasure more; put it in your hand.",
        "id": 1522,
        "image": "Mine.jpg",
        "name": "Mine",
        "spendingPower": 0,
        "tags": Array [],
      }
    }
    key="1522"
  />
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 0,
        "buyingPower": 0,
        "cardsToDraw": 0,
        "category": Array [
          "Action",
        ],
        "cost": 4,
        "desc": "Trash a card from your hand. Gain a card costing up to 2 treasure more than the trashed card.",
        "id": 3333,
        "image": "Remodel.jpg",
        "name": "Remodel",
        "spendingPower": 0,
        "tags": Array [],
      }
    }
    key="3333"
  />
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 1,
        "buyingPower": 0,
        "cardsToDraw": 0,
        "category": Array [
          "Action",
        ],
        "cost": 2,
        "desc": "Discard any number of cards. +1 Card per card discarded.",
        "id": 9831,
        "image": "Cellar.jpg",
        "name": "Cellar",
        "spendingPower": 0,
        "tags": Array [
          "+1 Action",
        ],
      }
    }
    key="9831"
  />
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 0,
        "buyingPower": 0,
        "cardsToDraw": 2,
        "category": Array [
          "Action",
          "Reaction",
        ],
        "cost": 2,
        "desc": "When another player plays an Attack card, you may reveal this from your hand. If you do, you are unaffected by that attack.",
        "id": 1938,
        "image": "Moat.jpg",
        "name": "Moat",
        "spendingPower": 0,
        "tags": Array [
          "+2 Cards",
        ],
      }
    }
    key="1938"
  />
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 0,
        "buyingPower": 1,
        "cardsToDraw": 0,
        "category": Array [
          "Action",
        ],
        "cost": 3,
        "desc": "",
        "id": 9781,
        "image": "WoodCutter.jpg",
        "name": "WoodCutter",
        "spendingPower": 2,
        "tags": Array [
          "+1 Buy",
          "+2 Gold",
        ],
      }
    }
    key="9781"
  />
  <Connect(ActionCard)
    card={
      Object {
        "actionsProvided": 0,
        "buyingPower": 0,
        "cardsToDraw": 0,
        "category": Array [
          "Action",
        ],
        "cost": 3,
        "desc": "Gain a card costing up to 4 treasure.",
        "id": 1333,
        "image": "Workshop.jpg",
        "name": "WorkShop",
        "spendingPower": 0,
        "tags": Array [],
      }
    }
    key="1333"
  />
  <Connect(Card)
    category={
      Array [
        "Victory",
      ]
    }
    cost={2}
    id={6123}
    image="Estate.jpg"
    key="6123"
    name="Estate"
  />
  <Connect(Card)
    category={
      Array [
        "Victory",
      ]
    }
    cost={5}
    id={1344}
    image="Duchy.jpg"
    key="1344"
    name="Duchy"
  />
  <Connect(Card)
    category={
      Array [
        "Victory",
      ]
    }
    cost={8}
    id={3781}
    image="Province.jpg"
    key="3781"
    name="Province"
  />
</section>
`;
